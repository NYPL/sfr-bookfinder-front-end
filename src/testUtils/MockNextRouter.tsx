import * as React from "react";
import Router from "next/router";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { NextRouter } from "next/router";

/**
 * Mock for the next/Router import.
 */

export const mockPush = jest.fn().mockImplementation(async () => true);
Router.push = mockPush;

export const MockNextRouterContextProvider: React.FC<{
  router?: Partial<NextRouter>;
}> = ({ router = {}, children }) => {
  const {
    basePath = "",
    route = "",
    pathname = "",
    query = {},
    asPath = "",
    push = mockPush,
    replace = jest.fn().mockImplementation(async () => true),
    reload = jest.fn().mockImplementation(() => null),
    back = jest.fn().mockImplementation(() => null),
    prefetch = jest.fn().mockImplementation(async () => undefined),
    beforePopState = jest.fn().mockImplementation(() => null),
    isFallback = false,
    events = {
      on: () => null,
      off: () => null,
      emit: () => null,
    },
  } = router;
  return (
    <RouterContext.Provider
      value={{
        basePath,
        route,
        pathname,
        query,
        asPath,
        push,
        replace,
        reload,
        back,
        prefetch,
        beforePopState,
        isFallback,
        events,
      }}
    >
      {children}
    </RouterContext.Provider>
  );
};

export default Router;
