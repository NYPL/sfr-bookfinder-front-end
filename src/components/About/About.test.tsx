import React from "react";
import { render } from "~/src/__tests__/testUtils/render";
import AboutPage from "./About";

it("renders about unchanged", async () => {
  const tree = render(<AboutPage />);
  expect(tree.container.firstChild).toMatchSnapshot();
});
