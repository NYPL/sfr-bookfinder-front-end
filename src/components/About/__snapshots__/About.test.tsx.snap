// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders about unchanged 1`] = `
<div
  className="css-1xdhyk6"
>
  <div
    className="css-0"
  >
    <nav
      aria-label="Breadcrumb"
      className="chakra-breadcrumb css-1xdhyk6"
    >
      <ol
        className="chakra-breadcrumb__list css-0"
      >
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <a
            className="chakra-breadcrumb__link css-0"
            href="https://www.nypl.org"
          >
            <span
              className="breadcrumb-label"
            >
              Home
            </span>
          </a>
          <span
            className="css-t4q1nq"
            role="presentation"
          >
            /
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <a
            className="chakra-breadcrumb__link css-0"
            href="https://www.nypl.org/research"
          >
            <span
              className="breadcrumb-label"
            >
              Research
            </span>
          </a>
          <span
            className="css-t4q1nq"
            role="presentation"
          >
            /
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <a
            className="chakra-breadcrumb__link css-0"
            href="/"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon breadcrumbs-icon css-1vgifmd"
              focusable={false}
              id="undefined__backarrow"
              role="img"
              viewBox="0 0 24 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                arrow icon
              </title>
              <path
                clipRule="evenodd"
                d="M10.526 12.871L.263 1.676 1.737.324 12 11.52 22.263.324l1.474 1.352L13.474 12.87a2 2 0 01-2.948 0z"
                fillRule="evenodd"
              />
            </svg>
            <span
              className="breadcrumb-label"
            >
              Digital Research Books Beta
            </span>
          </a>
          <span
            className="css-t4q1nq"
            role="presentation"
          >
            /
          </span>
        </li>
        <li
          className="chakra-breadcrumb__list-item css-18biwo"
        >
          <span
            aria-current="page"
            className="chakra-breadcrumb__link css-0"
          >
            <span
              className="breadcrumb-label"
            >
              About
            </span>
          </span>
        </li>
      </ol>
    </nav>
  </div>
  <main
    className="css-0"
    id="mainContent"
  >
    <div
      className="css-0"
    >
      <h1
        className="chakra-heading css-1xdhyk6"
      >
        <span>
          <span
            className="rn-section-title__emphasis"
          >
            Digital Research Books
          </span>
           
          Beta
        </span>
      </h1>
      <p>
        Digital Research Books Beta is an experimental project, now in early Beta testing, that collects digital versions of research books from many different sources, including Open Access publications, into one convenient place to search.
      </p>
      <p>
        All the materials in Digital Research Books Beta are completely free to read and most of them you can download and keep, with no library card required. The books are either in the
         
        <a
          className="css-1xdhyk6"
          href="/license"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          public domain
        </a>
        , with no restrictions on your use of them, or under
         
        <a
          className="css-1xdhyk6"
          href="/license"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          Creative Commons licences
        </a>
         that may have some conditions, but only on redistribution or adaptation.
      </p>
      <p>
        In addition to collecting these digital editions, we group all the editions of the same title together as a single “work.” For instance there are many editions of
         
        <a
          className="css-1xdhyk6"
          href="/work/e34d73df-f32b-49e1-8fdf-151db2a7806a"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          Mary Wollstonecraft’s A Vindication of the Rights of Woman
        </a>
        , many of them available digitally. We group them all together under a single search result and try to make the differences between them--years when and places where they were published, for instance--easy to understand.
      </p>
      <h2
        className="chakra-heading css-1xdhyk6"
        id="sources-and-data-heading"
      >
        Sources and Data
      </h2>
      <p>
        The material in Digital Research Books Beta are drawn from several public sources, mainly
         
        <a
          className="css-1xdhyk6"
          href="https://www.hathitrust.org/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          HathiTrust
        </a>
        ,
         
        <a
          className="css-1xdhyk6"
          href="https://doabooks.org/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          The Directory of Open Access Books
        </a>
        , and 
        <a
          className="css-1xdhyk6"
          href="http://www.gutenberg.org/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          Project Gutenberg
        </a>
        . We are continuously adding more books from these and other sources. We then cross-reference them with library records from NYPL and
         
        <a
          className="css-1xdhyk6"
          href="https://www.worldcat.org/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          WorldCat
        </a>
        , using OCLC’s experimental
         
        <a
          className="css-1xdhyk6"
          href="http://classify.oclc.org/classify2/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          rel={null}
          target={null}
        >
          Classify
        </a>
         to make connections between different editions of the same work.
      </p>
      <h2
        className="chakra-heading css-1xdhyk6"
        id="beta-testing-heading"
      >
        What does Beta Testing mean?
      </h2>
      <p>
        It means that this is a work-in-progress. We are constantly working on the interface and trying new ideas to deal with the data. If you visit repeatedly, sometimes it may look different. If you search for the same thing often, the results may change--hopefully for the better as we refine how the search works. There may be errors. The cross-referencing is automated and based on millions of library records including some that are inaccurate. Part of what we are exploring is how to detect and adjust mistakes, and how to make sure we don’t introduce any new ones.
      </p>
      <p>
        It also means that the project may change radically. We may change the URL. We may learn that a different approach is necessary. We may learn that it isn’t useful enough to anyone to continue. If you find books in Digital Research Books Beta that are especially useful to you, you should download a copy so that you have one no matter what becomes of this project.
      </p>
      <p>
        Most of all, it means your feedback is important! Most pages have a feedback button in the bottom right corner. We want to know what you think. If there are things you like or dislike, if there’s a feature missing, if you find an error please tell us in the feedback!
      </p>
    </div>
  </main>
</div>
`;
